// @trinsic/trinsic/lib/src/proto/services/verifiable-credentials/templates/v1/TemplatesServiceClientPb.js
"use strict";
/**
 * @fileoverview gRPC-Web generated client stub for services.verifiablecredentials.templates.v1
 * @enhanceable
 * @public
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CredentialTemplatesClient = void 0;
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
const grpcWeb = __importStar(require("grpc-web"));
const services_verifiable$credentials_templates_v1_templates_pb = __importStar(require("../../../../services/verifiable-credentials/templates/v1/templates_pb"));
class CredentialTemplatesClient {
    constructor(hostname, credentials, options) {
        this.methodInfoCreate = new grpcWeb.AbstractClientBase.MethodInfo(services_verifiable$credentials_templates_v1_templates_pb.CreateCredentialTemplateResponse, (request) => {
            return request.serializeBinary();
        }, services_verifiable$credentials_templates_v1_templates_pb.CreateCredentialTemplateResponse.deserializeBinary);
        this.methodInfoGet = new grpcWeb.AbstractClientBase.MethodInfo(services_verifiable$credentials_templates_v1_templates_pb.GetCredentialTemplateResponse, (request) => {
            return request.serializeBinary();
        }, services_verifiable$credentials_templates_v1_templates_pb.GetCredentialTemplateResponse.deserializeBinary);
        this.methodInfoList = new grpcWeb.AbstractClientBase.MethodInfo(services_verifiable$credentials_templates_v1_templates_pb.ListCredentialTemplatesResponse, (request) => {
            return request.serializeBinary();
        }, services_verifiable$credentials_templates_v1_templates_pb.ListCredentialTemplatesResponse.deserializeBinary);
        this.methodInfoSearch = new grpcWeb.AbstractClientBase.MethodInfo(services_verifiable$credentials_templates_v1_templates_pb.SearchCredentialTemplatesResponse, (request) => {
            return request.serializeBinary();
        }, services_verifiable$credentials_templates_v1_templates_pb.SearchCredentialTemplatesResponse.deserializeBinary);
        this.methodInfoDelete = new grpcWeb.AbstractClientBase.MethodInfo(services_verifiable$credentials_templates_v1_templates_pb.DeleteCredentialTemplateResponse, (request) => {
            return request.serializeBinary();
        }, services_verifiable$credentials_templates_v1_templates_pb.DeleteCredentialTemplateResponse.deserializeBinary);
        if (!options)
            options = {};
        if (!credentials)
            credentials = {};
        options["format"] = "text";
        this.client_ = new grpcWeb.GrpcWebClientBase(options);
        this.hostname_ = hostname;
        this.credentials_ = credentials;
        this.options_ = options;
    }
    create(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.verifiablecredentials.templates.v1.CredentialTemplates/Create", request, metadata || {}, this.methodInfoCreate, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.verifiablecredentials.templates.v1.CredentialTemplates/Create", request, metadata || {}, this.methodInfoCreate);
    }
    get(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.verifiablecredentials.templates.v1.CredentialTemplates/Get", request, metadata || {}, this.methodInfoGet, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.verifiablecredentials.templates.v1.CredentialTemplates/Get", request, metadata || {}, this.methodInfoGet);
    }
    list(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.verifiablecredentials.templates.v1.CredentialTemplates/List", request, metadata || {}, this.methodInfoList, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.verifiablecredentials.templates.v1.CredentialTemplates/List", request, metadata || {}, this.methodInfoList);
    }
    search(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.verifiablecredentials.templates.v1.CredentialTemplates/Search", request, metadata || {}, this.methodInfoSearch, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.verifiablecredentials.templates.v1.CredentialTemplates/Search", request, metadata || {}, this.methodInfoSearch);
    }
    delete(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.verifiablecredentials.templates.v1.CredentialTemplates/Delete", request, metadata || {}, this.methodInfoDelete, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.verifiablecredentials.templates.v1.CredentialTemplates/Delete", request, metadata || {}, this.methodInfoDelete);
    }
}
exports.CredentialTemplatesClient = CredentialTemplatesClient;
