// @trinsic/trinsic/lib/src/proto/services/universal-wallet/v1/Universal-walletServiceClientPb.js
"use strict";
/**
 * @fileoverview gRPC-Web generated client stub for services.universalwallet.v1
 * @enhanceable
 * @public
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UniversalWalletClient = void 0;
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
const grpcWeb = __importStar(require("grpc-web"));
const services_universal$wallet_v1_universal$wallet_pb = __importStar(require("../../../services/universal-wallet/v1/universal-wallet_pb"));
class UniversalWalletClient {
    constructor(hostname, credentials, options) {
        this.methodInfoGetItem = new grpcWeb.AbstractClientBase.MethodInfo(services_universal$wallet_v1_universal$wallet_pb.GetItemResponse, (request) => {
            return request.serializeBinary();
        }, services_universal$wallet_v1_universal$wallet_pb.GetItemResponse.deserializeBinary);
        this.methodInfoSearch = new grpcWeb.AbstractClientBase.MethodInfo(services_universal$wallet_v1_universal$wallet_pb.SearchResponse, (request) => {
            return request.serializeBinary();
        }, services_universal$wallet_v1_universal$wallet_pb.SearchResponse.deserializeBinary);
        this.methodInfoInsertItem = new grpcWeb.AbstractClientBase.MethodInfo(services_universal$wallet_v1_universal$wallet_pb.InsertItemResponse, (request) => {
            return request.serializeBinary();
        }, services_universal$wallet_v1_universal$wallet_pb.InsertItemResponse.deserializeBinary);
        this.methodInfoUpdateItem = new grpcWeb.AbstractClientBase.MethodInfo(services_universal$wallet_v1_universal$wallet_pb.UpdateItemResponse, (request) => {
            return request.serializeBinary();
        }, services_universal$wallet_v1_universal$wallet_pb.UpdateItemResponse.deserializeBinary);
        this.methodInfoDeleteItem = new grpcWeb.AbstractClientBase.MethodInfo(services_universal$wallet_v1_universal$wallet_pb.DeleteItemResponse, (request) => {
            return request.serializeBinary();
        }, services_universal$wallet_v1_universal$wallet_pb.DeleteItemResponse.deserializeBinary);
        if (!options)
            options = {};
        if (!credentials)
            credentials = {};
        options["format"] = "text";
        this.client_ = new grpcWeb.GrpcWebClientBase(options);
        this.hostname_ = hostname;
        this.credentials_ = credentials;
        this.options_ = options;
    }
    getItem(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.universalwallet.v1.UniversalWallet/GetItem", request, metadata || {}, this.methodInfoGetItem, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.universalwallet.v1.UniversalWallet/GetItem", request, metadata || {}, this.methodInfoGetItem);
    }
    search(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.universalwallet.v1.UniversalWallet/Search", request, metadata || {}, this.methodInfoSearch, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.universalwallet.v1.UniversalWallet/Search", request, metadata || {}, this.methodInfoSearch);
    }
    insertItem(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.universalwallet.v1.UniversalWallet/InsertItem", request, metadata || {}, this.methodInfoInsertItem, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.universalwallet.v1.UniversalWallet/InsertItem", request, metadata || {}, this.methodInfoInsertItem);
    }
    updateItem(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.universalwallet.v1.UniversalWallet/UpdateItem", request, metadata || {}, this.methodInfoUpdateItem, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.universalwallet.v1.UniversalWallet/UpdateItem", request, metadata || {}, this.methodInfoUpdateItem);
    }
    deleteItem(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.universalwallet.v1.UniversalWallet/DeleteItem", request, metadata || {}, this.methodInfoDeleteItem, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.universalwallet.v1.UniversalWallet/DeleteItem", request, metadata || {}, this.methodInfoDeleteItem);
    }
}
exports.UniversalWalletClient = UniversalWalletClient;
