// @trinsic/trinsic/lib/src/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceBase = exports.TemplateService = exports.CredentialService = exports.AccountService = exports.TrustRegistryService = exports.ProviderService = exports.WalletService = exports.TrinsicService = void 0;
const WalletService_1 = require("./WalletService");
Object.defineProperty(exports, "WalletService", { enumerable: true, get: function () { return WalletService_1.WalletService; } });
const ProviderService_1 = require("./ProviderService");
Object.defineProperty(exports, "ProviderService", { enumerable: true, get: function () { return ProviderService_1.ProviderService; } });
const TrustRegistryService_1 = require("./TrustRegistryService");
Object.defineProperty(exports, "TrustRegistryService", { enumerable: true, get: function () { return TrustRegistryService_1.TrustRegistryService; } });
const CredentialService_1 = require("./CredentialService");
Object.defineProperty(exports, "CredentialService", { enumerable: true, get: function () { return CredentialService_1.CredentialService; } });
const TemplateService_1 = require("./TemplateService");
Object.defineProperty(exports, "TemplateService", { enumerable: true, get: function () { return TemplateService_1.TemplateService; } });
const AccountService_1 = require("./AccountService");
Object.defineProperty(exports, "AccountService", { enumerable: true, get: function () { return AccountService_1.AccountService; } });
const TrinsicService_1 = require("./TrinsicService");
Object.defineProperty(exports, "TrinsicService", { enumerable: true, get: function () { return TrinsicService_1.TrinsicService; } });
const ServiceBase_1 = __importDefault(require("./ServiceBase"));
exports.ServiceBase = ServiceBase_1.default;
__exportStar(require("./proto/index"), exports);
