// @trinsic/trinsic/lib/src/TemplateService.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplateService = void 0;
const ServiceBase_1 = __importDefault(require("./ServiceBase"));
const proto_1 = require("./proto");
const proto = __importStar(require("./proto"));
class TemplateService extends ServiceBase_1.default {
    constructor(options) {
        super(options);
        this.client = this.createClient(proto_1.CredentialTemplatesDefinition);
    }
    // BEGIN Code generated by protoc-gen-trinsic. DO NOT EDIT.
    // target: /home/runner/work/sdk/sdk/web/src/TemplateService.ts
    /** Create a credential template in the current ecosystem */
    async create(request) {
        return this.client.create(request, {
            metadata: await this.buildMetadata(proto.CreateCredentialTemplateRequest.encode(request).finish())
        });
    }
    /** Fetch a credential template by ID */
    async get(request) {
        return this.client.get(request, {
            metadata: await this.buildMetadata(proto.GetCredentialTemplateRequest.encode(request).finish())
        });
    }
    /** Search credential templates using SQL, returning strongly-typed template data */
    async list(request) {
        return this.client.list(request, {
            metadata: await this.buildMetadata(proto.ListCredentialTemplatesRequest.encode(request).finish())
        });
    }
    /** Search credential templates using SQL, returning raw JSON data */
    async search(request) {
        return this.client.search(request, {
            metadata: await this.buildMetadata(proto.SearchCredentialTemplatesRequest.encode(request).finish())
        });
    }
    /** Delete a credential template from the current ecosystem by ID */
    async delete(request) {
        return this.client.delete(request, {
            metadata: await this.buildMetadata(proto.DeleteCredentialTemplateRequest.encode(request).finish())
        });
    }
}
exports.TemplateService = TemplateService;
