// @trinsic/trinsic/lib/src/TrinsicService.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrinsicService = void 0;
const ServiceBase_1 = __importDefault(require("./ServiceBase"));
const AccountService_1 = require("./AccountService");
const CredentialService_1 = require("./CredentialService");
const TemplateService_1 = require("./TemplateService");
const ProviderService_1 = require("./ProviderService");
const TrustRegistryService_1 = require("./TrustRegistryService");
const WalletService_1 = require("./WalletService");
class TrinsicService extends ServiceBase_1.default {
    constructor(options) {
        super(options);
    }
    account() {
        this._account = this._account || new AccountService_1.AccountService(this.options);
        this._account.options = this.options;
        return this._account;
    }
    credential() {
        this._credential =
            this._credential || new CredentialService_1.CredentialService(this.options);
        this._credential.options = this.options;
        return this._credential;
    }
    provider() {
        this._provider = this._provider || new ProviderService_1.ProviderService(this.options);
        this._provider.options = this.options;
        return this._provider;
    }
    template() {
        this._template = this._template || new TemplateService_1.TemplateService(this.options);
        this._template.options = this.options;
        return this._template;
    }
    trustRegistry() {
        this._trustRegistry =
            this._trustRegistry || new TrustRegistryService_1.TrustRegistryService(this.options);
        this._trustRegistry.options = this.options;
        return this._trustRegistry;
    }
    wallet() {
        this._wallet = this._wallet || new WalletService_1.WalletService(this.options);
        this._wallet.options = this.options;
        return this._wallet;
    }
}
exports.TrinsicService = TrinsicService;
