// @trinsic/trinsic/lib/src/WalletService.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WalletService = void 0;
const ServiceBase_1 = __importDefault(require("./ServiceBase"));
const proto_1 = require("./proto");
const proto = __importStar(require("./proto"));
class WalletService extends ServiceBase_1.default {
    constructor(options) {
        super(options);
        this.client = this.createClient(proto_1.UniversalWalletDefinition);
    }
    async searchWallet(request = proto_1.SearchRequest.fromPartial({
        query: "SELECT c.id, c.type, c.data FROM c OFFSET 0 LIMIT 100",
    })) {
        return this.search(request);
    }
    // BEGIN Code generated by protoc-gen-trinsic. DO NOT EDIT.
    // target: /home/runner/work/sdk/sdk/web/src/WalletService.ts
    /** Retrieve an item from the wallet with a given item identifier */
    async getItem(request) {
        return this.client.getItem(request, {
            metadata: await this.buildMetadata(proto.GetItemRequest.encode(request).finish())
        });
    }
    /** Search the wallet using a SQL syntax */
    async search(request) {
        return this.client.search(request, {
            metadata: await this.buildMetadata(proto.SearchRequest.encode(request).finish())
        });
    }
    /** Insert an item into the wallet */
    async insertItem(request) {
        return this.client.insertItem(request, {
            metadata: await this.buildMetadata(proto.InsertItemRequest.encode(request).finish())
        });
    }
    /** Update an item in the wallet */
    async updateItem(request) {
        return this.client.updateItem(request, {
            metadata: await this.buildMetadata(proto.UpdateItemRequest.encode(request).finish())
        });
    }
    /** Delete an item from the wallet permanently */
    async deleteItem(request) {
        return this.client.deleteItem(request, {
            metadata: await this.buildMetadata(proto.DeleteItemRequest.encode(request).finish())
        });
    }
}
exports.WalletService = WalletService;
