// @trinsic/okapi-web/lib/proto/okapi/examples/v1/examples.js
/* eslint-disable */
import { EncryptedMessage } from "../../../pbmse/v1/pbmse";
import _m0 from "protobufjs/minimal";
function createBaseBasicMessage() {
    return { text: "" };
}
export const BasicMessage = {
    encode(message, writer = _m0.Writer.create()) {
        if (message.text !== "") {
            writer.uint32(10).string(message.text);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0.Reader ? input : new _m0.Reader(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseBasicMessage();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    message.text = reader.string();
                    break;
                default:
                    reader.skipType(tag & 7);
                    break;
            }
        }
        return message;
    },
    fromJSON(object) {
        return {
            text: isSet(object.text) ? String(object.text) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.text !== undefined && (obj.text = message.text);
        return obj;
    },
    fromPartial(object) {
        var _a;
        const message = createBaseBasicMessage();
        message.text = (_a = object.text) !== null && _a !== void 0 ? _a : "";
        return message;
    },
};
export const SecureExampleServiceDefinition = {
    name: "SecureExampleService",
    fullName: "okapi.examples.v1.SecureExampleService",
    methods: {
        unary: {
            name: "Unary",
            requestType: EncryptedMessage,
            requestStream: false,
            responseType: EncryptedMessage,
            responseStream: false,
            options: {},
        },
        serverStreaming: {
            name: "ServerStreaming",
            requestType: EncryptedMessage,
            requestStream: false,
            responseType: EncryptedMessage,
            responseStream: true,
            options: {},
        },
    },
};
function isSet(value) {
    return value !== null && value !== undefined;
}
