// @trinsic/trinsic/lib/src/proto/services/verifiable-credentials/v1/Verifiable-credentialsServiceClientPb.js
"use strict";
/**
 * @fileoverview gRPC-Web generated client stub for services.verifiablecredentials.v1
 * @enhanceable
 * @public
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerifiableCredentialClient = void 0;
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
const grpcWeb = __importStar(require("grpc-web"));
const services_verifiable$credentials_v1_verifiable$credentials_pb = __importStar(require("../../../services/verifiable-credentials/v1/verifiable-credentials_pb"));
class VerifiableCredentialClient {
    constructor(hostname, credentials, options) {
        this.methodInfoIssue = new grpcWeb.AbstractClientBase.MethodInfo(services_verifiable$credentials_v1_verifiable$credentials_pb.IssueResponse, (request) => {
            return request.serializeBinary();
        }, services_verifiable$credentials_v1_verifiable$credentials_pb.IssueResponse.deserializeBinary);
        this.methodInfoIssueFromTemplate = new grpcWeb.AbstractClientBase.MethodInfo(services_verifiable$credentials_v1_verifiable$credentials_pb.IssueFromTemplateResponse, (request) => {
            return request.serializeBinary();
        }, services_verifiable$credentials_v1_verifiable$credentials_pb.IssueFromTemplateResponse.deserializeBinary);
        this.methodInfoCheckStatus = new grpcWeb.AbstractClientBase.MethodInfo(services_verifiable$credentials_v1_verifiable$credentials_pb.CheckStatusResponse, (request) => {
            return request.serializeBinary();
        }, services_verifiable$credentials_v1_verifiable$credentials_pb.CheckStatusResponse.deserializeBinary);
        this.methodInfoUpdateStatus = new grpcWeb.AbstractClientBase.MethodInfo(services_verifiable$credentials_v1_verifiable$credentials_pb.UpdateStatusResponse, (request) => {
            return request.serializeBinary();
        }, services_verifiable$credentials_v1_verifiable$credentials_pb.UpdateStatusResponse.deserializeBinary);
        this.methodInfoCreateProof = new grpcWeb.AbstractClientBase.MethodInfo(services_verifiable$credentials_v1_verifiable$credentials_pb.CreateProofResponse, (request) => {
            return request.serializeBinary();
        }, services_verifiable$credentials_v1_verifiable$credentials_pb.CreateProofResponse.deserializeBinary);
        this.methodInfoVerifyProof = new grpcWeb.AbstractClientBase.MethodInfo(services_verifiable$credentials_v1_verifiable$credentials_pb.VerifyProofResponse, (request) => {
            return request.serializeBinary();
        }, services_verifiable$credentials_v1_verifiable$credentials_pb.VerifyProofResponse.deserializeBinary);
        this.methodInfoSend = new grpcWeb.AbstractClientBase.MethodInfo(services_verifiable$credentials_v1_verifiable$credentials_pb.SendResponse, (request) => {
            return request.serializeBinary();
        }, services_verifiable$credentials_v1_verifiable$credentials_pb.SendResponse.deserializeBinary);
        if (!options)
            options = {};
        if (!credentials)
            credentials = {};
        options["format"] = "text";
        this.client_ = new grpcWeb.GrpcWebClientBase(options);
        this.hostname_ = hostname;
        this.credentials_ = credentials;
        this.options_ = options;
    }
    issue(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.verifiablecredentials.v1.VerifiableCredential/Issue", request, metadata || {}, this.methodInfoIssue, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.verifiablecredentials.v1.VerifiableCredential/Issue", request, metadata || {}, this.methodInfoIssue);
    }
    issueFromTemplate(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.verifiablecredentials.v1.VerifiableCredential/IssueFromTemplate", request, metadata || {}, this.methodInfoIssueFromTemplate, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.verifiablecredentials.v1.VerifiableCredential/IssueFromTemplate", request, metadata || {}, this.methodInfoIssueFromTemplate);
    }
    checkStatus(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.verifiablecredentials.v1.VerifiableCredential/CheckStatus", request, metadata || {}, this.methodInfoCheckStatus, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.verifiablecredentials.v1.VerifiableCredential/CheckStatus", request, metadata || {}, this.methodInfoCheckStatus);
    }
    updateStatus(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.verifiablecredentials.v1.VerifiableCredential/UpdateStatus", request, metadata || {}, this.methodInfoUpdateStatus, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.verifiablecredentials.v1.VerifiableCredential/UpdateStatus", request, metadata || {}, this.methodInfoUpdateStatus);
    }
    createProof(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.verifiablecredentials.v1.VerifiableCredential/CreateProof", request, metadata || {}, this.methodInfoCreateProof, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.verifiablecredentials.v1.VerifiableCredential/CreateProof", request, metadata || {}, this.methodInfoCreateProof);
    }
    verifyProof(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.verifiablecredentials.v1.VerifiableCredential/VerifyProof", request, metadata || {}, this.methodInfoVerifyProof, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.verifiablecredentials.v1.VerifiableCredential/VerifyProof", request, metadata || {}, this.methodInfoVerifyProof);
    }
    send(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.verifiablecredentials.v1.VerifiableCredential/Send", request, metadata || {}, this.methodInfoSend, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            "/services.verifiablecredentials.v1.VerifiableCredential/Send", request, metadata || {}, this.methodInfoSend);
    }
}
exports.VerifiableCredentialClient = VerifiableCredentialClient;
