// @trinsic/trinsic/lib/src/proto/services/debug/v1/DebugServiceClientPb.js
"use strict";
/**
 * @fileoverview gRPC-Web generated client stub for services.debug.v1
 * @enhanceable
 * @public
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DebuggingClient = void 0;
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
const grpcWeb = __importStar(require("grpc-web"));
const google_protobuf_empty_pb = __importStar(require("google-protobuf/google/protobuf/empty_pb"));
class DebuggingClient {
    constructor(hostname, credentials, options) {
        this.methodInfoCallEmpty = new grpcWeb.AbstractClientBase.MethodInfo(google_protobuf_empty_pb.Empty, (request) => {
            return request.serializeBinary();
        }, google_protobuf_empty_pb.Empty.deserializeBinary);
        this.methodInfoCallEmptyAuth = new grpcWeb.AbstractClientBase.MethodInfo(google_protobuf_empty_pb.Empty, (request) => {
            return request.serializeBinary();
        }, google_protobuf_empty_pb.Empty.deserializeBinary);
        if (!options)
            options = {};
        if (!credentials)
            credentials = {};
        options["format"] = "text";
        this.client_ = new grpcWeb.GrpcWebClientBase(options);
        this.hostname_ = hostname;
        this.credentials_ = credentials;
        this.options_ = options;
    }
    callEmpty(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ + "/services.debug.v1.Debugging/CallEmpty", request, metadata || {}, this.methodInfoCallEmpty, callback);
        }
        return this.client_.unaryCall(this.hostname_ + "/services.debug.v1.Debugging/CallEmpty", request, metadata || {}, this.methodInfoCallEmpty);
    }
    callEmptyAuth(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ + "/services.debug.v1.Debugging/CallEmptyAuth", request, metadata || {}, this.methodInfoCallEmptyAuth, callback);
        }
        return this.client_.unaryCall(this.hostname_ + "/services.debug.v1.Debugging/CallEmptyAuth", request, metadata || {}, this.methodInfoCallEmptyAuth);
    }
}
exports.DebuggingClient = DebuggingClient;
