// @trinsic/trinsic/lib/src/proto/services/provider/v1/ProviderServiceClientPb.js
"use strict";
/**
 * @fileoverview gRPC-Web generated client stub for services.provider.v1
 * @enhanceable
 * @public
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProviderClient = void 0;
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
const grpcWeb = __importStar(require("grpc-web"));
const services_provider_v1_provider_pb = __importStar(require("../../../services/provider/v1/provider_pb"));
class ProviderClient {
    constructor(hostname, credentials, options) {
        this.methodInfoCreateEcosystem = new grpcWeb.AbstractClientBase.MethodInfo(services_provider_v1_provider_pb.CreateEcosystemResponse, (request) => {
            return request.serializeBinary();
        }, services_provider_v1_provider_pb.CreateEcosystemResponse.deserializeBinary);
        this.methodInfoGenerateToken = new grpcWeb.AbstractClientBase.MethodInfo(services_provider_v1_provider_pb.GenerateTokenResponse, (request) => {
            return request.serializeBinary();
        }, services_provider_v1_provider_pb.GenerateTokenResponse.deserializeBinary);
        this.methodInfoInvite = new grpcWeb.AbstractClientBase.MethodInfo(services_provider_v1_provider_pb.InviteResponse, (request) => {
            return request.serializeBinary();
        }, services_provider_v1_provider_pb.InviteResponse.deserializeBinary);
        this.methodInfoInvitationStatus = new grpcWeb.AbstractClientBase.MethodInfo(services_provider_v1_provider_pb.InvitationStatusResponse, (request) => {
            return request.serializeBinary();
        }, services_provider_v1_provider_pb.InvitationStatusResponse.deserializeBinary);
        if (!options)
            options = {};
        if (!credentials)
            credentials = {};
        options["format"] = "text";
        this.client_ = new grpcWeb.GrpcWebClientBase(options);
        this.hostname_ = hostname;
        this.credentials_ = credentials;
        this.options_ = options;
    }
    createEcosystem(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.provider.v1.Provider/CreateEcosystem", request, metadata || {}, this.methodInfoCreateEcosystem, callback);
        }
        return this.client_.unaryCall(this.hostname_ + "/services.provider.v1.Provider/CreateEcosystem", request, metadata || {}, this.methodInfoCreateEcosystem);
    }
    generateToken(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ + "/services.provider.v1.Provider/GenerateToken", request, metadata || {}, this.methodInfoGenerateToken, callback);
        }
        return this.client_.unaryCall(this.hostname_ + "/services.provider.v1.Provider/GenerateToken", request, metadata || {}, this.methodInfoGenerateToken);
    }
    invite(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ + "/services.provider.v1.Provider/Invite", request, metadata || {}, this.methodInfoInvite, callback);
        }
        return this.client_.unaryCall(this.hostname_ + "/services.provider.v1.Provider/Invite", request, metadata || {}, this.methodInfoInvite);
    }
    invitationStatus(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                "/services.provider.v1.Provider/InvitationStatus", request, metadata || {}, this.methodInfoInvitationStatus, callback);
        }
        return this.client_.unaryCall(this.hostname_ + "/services.provider.v1.Provider/InvitationStatus", request, metadata || {}, this.methodInfoInvitationStatus);
    }
}
exports.ProviderClient = ProviderClient;
