// @trinsic/trinsic/lib/src/proto/services/account/v1/AccountServiceClientPb.js
"use strict";
/**
 * @fileoverview gRPC-Web generated client stub for services.account.v1
 * @enhanceable
 * @public
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountClient = void 0;
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
const grpcWeb = __importStar(require("grpc-web"));
const services_account_v1_account_pb = __importStar(require("../../../services/account/v1/account_pb"));
class AccountClient {
    constructor(hostname, credentials, options) {
        this.methodInfoSignIn = new grpcWeb.AbstractClientBase.MethodInfo(services_account_v1_account_pb.SignInResponse, (request) => {
            return request.serializeBinary();
        }, services_account_v1_account_pb.SignInResponse.deserializeBinary);
        this.methodInfoInfo = new grpcWeb.AbstractClientBase.MethodInfo(services_account_v1_account_pb.InfoResponse, (request) => {
            return request.serializeBinary();
        }, services_account_v1_account_pb.InfoResponse.deserializeBinary);
        this.methodInfoListDevices = new grpcWeb.AbstractClientBase.MethodInfo(services_account_v1_account_pb.ListDevicesResponse, (request) => {
            return request.serializeBinary();
        }, services_account_v1_account_pb.ListDevicesResponse.deserializeBinary);
        this.methodInfoRevokeDevice = new grpcWeb.AbstractClientBase.MethodInfo(services_account_v1_account_pb.RevokeDeviceResponse, (request) => {
            return request.serializeBinary();
        }, services_account_v1_account_pb.RevokeDeviceResponse.deserializeBinary);
        if (!options)
            options = {};
        if (!credentials)
            credentials = {};
        options["format"] = "text";
        this.client_ = new grpcWeb.GrpcWebClientBase(options);
        this.hostname_ = hostname;
        this.credentials_ = credentials;
        this.options_ = options;
    }
    signIn(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ + "/services.account.v1.Account/SignIn", request, metadata || {}, this.methodInfoSignIn, callback);
        }
        return this.client_.unaryCall(this.hostname_ + "/services.account.v1.Account/SignIn", request, metadata || {}, this.methodInfoSignIn);
    }
    info(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ + "/services.account.v1.Account/Info", request, metadata || {}, this.methodInfoInfo, callback);
        }
        return this.client_.unaryCall(this.hostname_ + "/services.account.v1.Account/Info", request, metadata || {}, this.methodInfoInfo);
    }
    listDevices(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ + "/services.account.v1.Account/ListDevices", request, metadata || {}, this.methodInfoListDevices, callback);
        }
        return this.client_.unaryCall(this.hostname_ + "/services.account.v1.Account/ListDevices", request, metadata || {}, this.methodInfoListDevices);
    }
    revokeDevice(request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ + "/services.account.v1.Account/RevokeDevice", request, metadata || {}, this.methodInfoRevokeDevice, callback);
        }
        return this.client_.unaryCall(this.hostname_ + "/services.account.v1.Account/RevokeDevice", request, metadata || {}, this.methodInfoRevokeDevice);
    }
}
exports.AccountClient = AccountClient;
